modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -86,36 +86,46 @@ export async function createCombinedKey(params: {
  * Prepares a NShare to be exchanged with other key holders.
  * Output is in a format that is usable within BitGo's ecosystem.
  *
- * @param params.keyShare - TSS key share of the party preparing exchange materials
- * @param params.recipientIndex - index of the recipient (1, 2, or 3)
- * @param params.recipientGpgPublicArmor - recipient's public gpg key in armor format
- * @param params.senderGpgPrivateArmor - sender's private gpg key in armor format
+ * @param keyShare - TSS key share of the party preparing exchange materials
+ * @param recipientIndex - index of the recipient (1, 2, or 3)
+ * @param recipientGpgPublicArmor - recipient's public gpg key in armor format
+ * @param senderGpgPrivateArmor - sender's private gpg key in armor format
  * @returns { EncryptedNShare } encrypted Y Share
  */
-export async function encryptNShare(params: {
-  keyShare: KeyShare;
-  recipientIndex: number;
-  recipientGpgPublicArmor: string;
-  senderGpgPrivateArmor: string;
-}): Promise<EncryptedNShare> {
-  const { keyShare, recipientIndex, recipientGpgPublicArmor, senderGpgPrivateArmor } = params;
-
+export async function encryptNShare(
+  keyShare: KeyShare,
+  recipientIndex: number,
+  recipientGpgPublicArmor: string,
+  senderGpgPrivateArmor: string
+): Promise<EncryptedNShare> {
   const nShare = keyShare.nShares[recipientIndex];
   if (!nShare) {
     throw new Error('Invalid recipient');
   }
 
-  const publicShare = keyShare.pShare.y;
-  const privateShare = nShare.u;
+  const publicShare = trimHexFromString(keyShare.pShare.y) + trimHexFromString(nShare.chaincode);
+  const privateShare = trimHexFromString(nShare.u) + trimHexFromString(nShare.n);
```

Comment:
I think n is not something which needs to be encrypted (Its paillier constant n right?), let's add it into the public share may be